pkg/controller/controller_test.go

```diff
@@ -180,12 +181,14 @@ func waitForStatefulSets(
 	cluster *myspec.M3DBCluster,
 	verb string,
 	opts waitForStatefulSetsOptions,
-) (waitForStatefulSetsResult, bool) {
+	until func(waitForStatefulSetsResult) bool,
+) waitForStatefulSetsResult {
```

Comment:
since the indexer can take "awhile" to update, 10 iterations of the reconcile loop might not be enough. Instead of increasing this arbitrary number further, I changed the logic to provide condition to wait until.